sklearn/utils/estimator_checks.py

```diff
@@ -1705,3 +1705,36 @@ def check_classifiers_regression_target(name, Estimator):
     e = Estimator()
     msg = 'Unknown label type: '
     assert_raises_regex(ValueError, msg, e.fit, X, y)
+
+
+@ignore_warnings(category=DeprecationWarning)
+def check_decision_proba_consistency(name, Estimator):
+    """
+    Check whether an estimator having both decision_function and
+    predict_proba methods has outputs with perfect rank correlation.
+    """
+
+    X, Y = make_multilabel_classification(n_classes=2, n_labels=1,
+                                          allow_unlabeled=True,
+                                          random_state=1)
+
+    estimator = Estimator()
+
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "decision_function"):
+
+        if hasattr(estimator, "predict_proba"):
+            try:
+                classif = OneVsRestClassifier(estimator)
+                classif.fit(X, Y)
+                a = classif.predict_proba([i for i in range(20)])
+                b = classif.decision_function([i for i in range(20)])
+                assert_equal(
+                 rankdata(a, method='average'), rankdata(b, method='average'))
+
+            except ValueError:
```

Comment:
What is this to catch? The `try` block should go around the smallest scope that we want to exclude otherwise this test can pass when all estimators raise a `ValueError` because the test is broken.